stablehlo/reference/Element.cpp

```diff
@@ -105,34 +104,73 @@ Element mapWithUpcastToDouble(const Element &el, FloatFn floatFn,
                               ComplexFn complexFn) {
   Type type = el.getType();
 
+  if (isSupportedFloatType(type))
+    return Element(type, floatFn(el.getFloatValue().convertToDouble()));
+
+  if (isSupportedComplexType(type))
+    return Element(type, complexFn(std::complex<double>(
+                             el.getComplexValue().real().convertToDouble(),
+                             el.getComplexValue().imag().convertToDouble())));
+
+  report_fatal_error(invalidArgument("Unsupported element type: %s",
+                                     debugString(type).c_str()));
+}
+
+}  // namespace
+
+Element::Element(Type type, APInt value) : type_(type), value_(value) {}
+
+Element::Element(Type type, int64_t value) {
+  type_ = type;
+  if (isSupportedSignedIntegerType(type)) {
+    value_ = APInt(type.getIntOrFloatBitWidth(), value, /*isSigned=*/true);
+  } else if (isSupportedUnsignedIntegerType(type)) {
+    value_ = APInt(type.getIntOrFloatBitWidth(), value, /*isSigned=*/false);
+  } else {
+    report_fatal_error(invalidArgument("Unsupported element type: %s",
+                                       debugString(type).c_str()));
+  }
+}
+
+Element::Element(Type type, bool value) : type_(type), value_(value) {}
+
+Element::Element(Type type, APFloat value) : type_(type), value_(value) {}
+
+Element::Element(Type type, double value) {
+  type_ = type;
   if (isSupportedFloatType(type)) {
-    APFloat elVal = el.getFloatValue();
-    const llvm::fltSemantics &elSemantics = elVal.getSemantics();
-    APFloat resultVal(floatFn(elVal.convertToDouble()));
+    APFloat floatVal(static_cast<double>(value));
```

Comment:
This is more of a preference question, so feel free to keep it open.

How  about we avoid if/else if possible. To me (and here comes the preference) it seems more cleaner than `if {} else {}`. 

```
if(!isSupportedFloatType())
  report_fatal_error

APFloat floatVal(static_cast<double>(value));
 bool roundingErr;
floatVal.convert(type.cast<FloatType>().getFloatSemantics(),
                     APFloat::rmNearestTiesToEven, &roundingErr);
 value_ = floatVal;
```

same below.